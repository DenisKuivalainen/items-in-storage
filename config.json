{
    "categories": ["gloves", "facemasks", "beanies"],

    "api": {
        "products": {
            "url": "https://bad-api-assignment.reaktor.com/v2/products/",
            "params": {
                "headers": {
                    "x-force-error-mode": "all"
                }
            }
        },
        "availability": {
            "url": "https://bad-api-assignment.reaktor.com/v2/availability/",
            "params": {
                "headers": {
                    "x-force-error-mode": "all"
                }
            }
        }
    },

    "server": {
        "url": "https://reactorjuniortask.herokuapp.com/"
    }
}