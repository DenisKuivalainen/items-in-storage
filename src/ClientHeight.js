const { useState, useLayoutEffect } = require("react");

const useClientHeight = (sub) => {
    const [size, setSize] = useState(0);
    const updateSize = () => setSize(document.documentElement.clientHeight);

    // Get window size
    useLayoutEffect(() => {
        window.addEventListener('resize', updateSize);
        updateSize();
        return () => window.removeEventListener('resize', updateSize);
    }, []);

    // Get client size
    sub = !!sub ? sub : 0;

    return size - sub;
}

module.exports = { useClientHeight };