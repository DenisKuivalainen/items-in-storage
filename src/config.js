const { __ } = require("ramda");
const { viewOnPath } = require("ramda-godlike");
const configFile = require("../config.json");

const config = (...args) => viewOnPath(args, configFile);

module.exports = { config };