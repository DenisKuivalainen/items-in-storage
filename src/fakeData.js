const { times, identity, map, toString, splitEvery, join, compose, curry } = require("ramda");

const getValue = curry((n, l) => {
    let letters = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"];

    return compose(
        x => l + "-" + x,
        join(""),
        map(x => letters[parseInt(x)]),
        splitEvery(1),
        toString
    )(n)

});

const getObject = (val) => {
    const gv = getValue(val);

    return { "id": val, "type": gv("tp"), "name": gv("nm"), "color": gv("cl"), "price": ((val + 254) * 3) + "$", "manufacturer": gv("mn"), "availability": gv("av") }
}

const data = (n) => map(getObject, times(identity, n));

module.exports = { data };