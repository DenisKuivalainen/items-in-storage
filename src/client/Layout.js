import React from 'react';
import { createMuiTheme } from "@material-ui/core/styles";
import { ThemeProvider } from "@material-ui/styles";
import Nav from "./Nav";
import { config } from "../config";
import { useClientHeight } from "../ClientHeight";

const Layout = ({ children }) => (
    <ThemeProvider theme={createMuiTheme({palette: {type: 'dark'}})}>
        <div className="index wrapper" style={{height: useClientHeight(20)}}>
            {Nav(config("categories"))}
            {children}
        </div>
    </ThemeProvider>
);

export default Layout;