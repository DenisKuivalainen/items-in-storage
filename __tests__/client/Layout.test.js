import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../../src/client/Layout';
import { config } from '../../src/config';
import { useRouter } from "next/router";

jest.mock('../../src/config');
config.mockReturnValue(["boots", "hoodies"]);

jest.mock("next/router");
useRouter.mockImplementation(() => { return {push: () => {}, pathname: "boots"} });

describe("test Layout ", () => {
    test("sanpshot for categories 'boots' and 'hoodies' with a children saying 'sup'", () => {
        const tree = renderer.create(<Layout><p>sup</p></Layout>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})