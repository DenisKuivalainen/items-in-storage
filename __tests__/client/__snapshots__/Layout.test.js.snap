// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Layout  sanpshot for categories 'boots' and 'hoodies' with a children saying 'sup' 1`] = `
<div
  className="index wrapper"
  style={
    Object {
      "height": -20,
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        href="/boots"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          boots
        </span>
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
        href="/hoodies"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "width": "100%",
          }
        }
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          hoodies
        </span>
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    />
  </div>
  <p>
    sup
  </p>
</div>
`;
